pkg/state/script_caller.go

```diff
@@ -187,7 +189,9 @@ func (a *scriptCaller) callAssetScriptCommon(env *ride.EvaluationEnvironment, se
 		}
 		env.SetThisFromFullAssetInfo(assetInfo)
 	}
-	env.SetLastBlock(params.blockInfo)
+	if err := env.SetLastBlockFromBlockInfo(params.blockInfo); err != nil {
+		return nil, err
```

Comment:
Maybe add info for the error?

Reply:
No error wrapping before the script execution in this function, so it's the same.